ambry-tools/src/main/java/com.github.ambry/store/DumpDataHelper.java

```diff
@@ -260,6 +265,9 @@ LogBlobRecordInfo readSingleRecordFromLog(RandomAccessFile randomAccessFile, lon
     } else {
       throw new MessageFormatException("Header version not supported " + version, MessageFormatErrorCodes.IO_Error);
     }
+    if(throttler != null){
+      throttler.maybeThrottle(totalRecordSize);
```

Comment:
Yes, its not straight forward. For instance, totalRecordSize refers to the totalbytes read as part of this blob. Which is nothing by the totalObservedUnit or totalObservedBytes from last time.